File: javascript/src/models/appUsageStatsIn.ts
// this file is @generated

export interface AppUsageStatsIn {
  since: Date;
  until: Date;
  /**
   * Specific app IDs or UIDs to aggregate stats for.
   *
   * Note that if none of the given IDs or UIDs are resolved, a 422 response will be given.
   */
  appIds?: string[] | null;
}

export const AppUsageStatsInSerializer = {
  _fromJsonObject(object: any): AppUsageStatsIn {
    return {
      since: new Date(object["since"]),
      until: new Date(object["until"]),
      appIds: object["appIds"],
    };
  },

  _toJsonObject(self: AppUsageStatsIn): any {
    return {
      since: self.since,
      until: self.until,
      appIds: self.appIds,
    };
  },
};

File: rust/src/models/app_usage_stats_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct AppUsageStatsIn {
    pub since: chrono::DateTime<chrono::Utc>,

    pub until: chrono::DateTime<chrono::Utc>,

    /// Specific app IDs or UIDs to aggregate stats for.
    ///
    /// Note that if none of the given IDs or UIDs are resolved, a 422 response
    /// will be given.
    #[serde(rename = "appIds")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub app_ids: Option<std::collections::BTreeSet<String>>,
}

impl AppUsageStatsIn {
    pub fn new(since: chrono::DateTime<chrono::Utc>, until: chrono::DateTime<chrono::Utc>) -> Self {
        Self {
            since,
            until,
            app_ids: None,
        }
    }
}
